mirrord/protocol/src/file.rs

```diff
@@ -385,11 +392,31 @@ pub struct ReadDirRequest {
     pub remote_fd: u64,
 }
 
+/// `readdir` message that requests an iterable with `amount` items from the agent.
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub struct ReadDirBatchRequest {
+    /// The fd of the dir in the agent.
+    pub remote_fd: u64,
+    /// Max amount to take from the agent's iterator of dirs.
+    pub amount: usize,
+}
+
 #[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
 pub struct ReadDirResponse {
     pub direntry: Option<DirEntryInternal>,
 }
 
+/// `readdir` response with the list of items (length depends on the [`ReadDirBatchRequest`]'s
+/// `amount`), and the `remote_fd` of the dir (for convenience).
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub struct ReadDirBatchResponse {
+    /// Remote fd of the dir.
+    pub fd: u64,
```

Comment:
What do we need this for? Agent handles file requests sequentially, so file responses don't usually contain the descriptor. Just a question though, I don't mind if it stays here

Reply:
Was the easiest way I found to updating the iterator:

```rust
                    if let Some(FileResource::Dir { dirs_iter }) =
                        self.remote_fds.get_mut(&layer_id, &RemoteFd::Dir(fd))
                    {
                        *dirs_iter = entries_iter;
                    }

```

Reply:
Ah, okok